src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,29 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
Using Executors.newCachedThreadPool() can lead to performance issues if a large number of threads are created and left idle. Consider using a fixed thread pool or another type of ExecutorService that better suits your needs, especially if you expect a high volume of events.